drivers/include/si1145.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2016 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_si1145 SI1145
+ * @ingroup     drivers_sensors
+ * @brief       Device driver interface for the SI1145 sensor
+ * @{
+ *
+ * @file
+ * @brief       Device driver interface for the SI1145 sensor.
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef SI1145_H_
+#define SI1145_H_
+
+#include "saul.h"
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Device descriptor for the SI1145 sensor
+ */
+typedef struct {
+    i2c_t i2c_dev;                     /**< I2C device which is used */
+} si1145_t;
+
+
+/**
+ * @brief Device initialization parameters
+ */
+typedef struct {
+    i2c_t i2c_dev;
+} si1145_params_t;
+
+/**
+ * @brief export SAUL endpoints
+ * @{
+ */
+extern const saul_driver_t si1145_uv_saul_driver;
+extern const saul_driver_t si1145_ir_saul_driver;
+extern const saul_driver_t si1145_visible_saul_driver;
+extern const saul_driver_t si1145_distance_saul_driver;
+/** @} */
+
+/**
+ * @brief auto-initialize all configured SI1145 devices
+ */
+void si1145_auto_init(void);
```

Comment:
So there are two ways for SAUL devices to auto init: either as part of `sys/auto_init/saul`, or in the driver itself. Not sure what is the preferred way?

I did it different for the Si70xx, which I found 'cleaner'.